From 6aa6dc215b2f7edbd77a302ead90b85ae481d253 Mon Sep 17 00:00:00 2001
Message-Id: <6aa6dc215b2f7edbd77a302ead90b85ae481d253@dist-git>
From: Erik Skultety <eskultet@redhat.com>
Date: Mon, 1 Aug 2016 16:40:53 +0200
Subject: [PATCH] virt-admin: Output srv-threadpool-info data as unsigned int
 rather than signed

Internally, all the data are represented as unsigned int, it is also documented
in the header file that users should use our exported constants that also
indicate that the data should be unsigned int. However, when polling for the
current server threadpool's configuration, virt-admin uses an incorrect
formatting parameter '%d' for printf. Instead, virt-admin should use formatting
parameter '%u'.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1356769

Signed-off-by: Erik Skultety <eskultet@redhat.com>
(cherry picked from commit a8962f705d5b7973e322b81c38e46eb542cb1ad4)
Signed-off-by: Erik Skultety <eskultet@redhat.com>
---
 tools/virt-admin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/virt-admin.c b/tools/virt-admin.c
index c4ee8f5..a59c4c7 100644
--- a/tools/virt-admin.c
+++ b/tools/virt-admin.c
@@ -458,7 +458,7 @@ cmdSrvThreadpoolInfo(vshControl *ctl, const vshCmd *cmd)
     }
 
     for (i = 0; i < nparams; i++)
-        vshPrint(ctl, "%-15s: %d\n", params[i].field, params[i].value.ui);
+        vshPrint(ctl, "%-15s: %u\n", params[i].field, params[i].value.ui);
 
     ret = true;
 
-- 
2.9.2

